src/frontends/common/include/openvino/frontend/frontend.hpp

```diff
@@ -20,12 +20,21 @@ namespace frontend {
 /// \brief An interface for identifying a frontend for a particular framework.
 /// Provides an ability to load and convert of input model
 class FRONTEND_API FrontEnd {
+    friend class FrontEndManager;
+    std::shared_ptr<void> m_shared_object = {};  // Library handle
+    std::shared_ptr<FrontEnd> m_actual = {};
+
 public:
-    typedef std::shared_ptr<FrontEnd> Ptr;
+    using Ptr = std::shared_ptr<FrontEnd>;
+
+    /// \brief Constructor holding shared object and actual FrontEnd's implementation.
+    /// Used by FrontEndManager to create FrontEnd object in core runtime
+    FrontEnd(const std::shared_ptr<void>& so, const std::shared_ptr<FrontEnd>& actual);
```

Comment:
why is it not a private?

Reply:
This was because std::make_shared(args) doesn't work for private class constructor even if caller is 'friend'. I've changed this to 'std::make_shared<>()' and then filling m_so and m_actual manually.
Another options I tried: a) create object with `auto* fe=new(so, actual);` then wrap it with 'std::unique_ptr', then create `std::shared_ptr(std::move(unique_ptr))` - this was also looked not pretty b) Define in frontendmanager.cpp `class FrontEndDerived: public FrontEnd` with public constructor, and call it. This also worked but looked strange, and additionally FrontEnd shall be a friend of 'InputModelDerived: public InputModel' and it will be stated in public frontend.hpp.
So I've chosen create std::make_shared without arguments and then fill members manually.